chillTime.py

```diff
@@ -79,60 +73,57 @@ def get_scalar(self, obs_type, record, db_manager):
         # Get the first element of the ValueTuple. This will be in Celsius:
         outTemp_F = outTemp_F_vt[0]
         interval_H = interval_H_vt[0]
-        
+
         if self.algorithm == 'simple':
             # Use the "Simple" algorithm. Results will be in hours. Create a ValueTuple out of it:
             if outTemp_F < 45:
                 chill_time = interval_H
             else:
                 chill_time = 0
-            chill_vt = ValueTuple(chill_time, 'hour', 'group_elapsed')
             log.debug("Found Simple chill hours of %s hours", chill_time)
         elif self.algorithm == 'utah':
             # Use the "Utah" algorithm, weighting chill temps between 32 and 45F
             if outTemp_F <= 34:
                 chill_time = 0
-            elif 34 < outTemp_F and outTemp_F <= 36:
+            elif 34 < outTemp_F <= 36:
                 chill_time = interval_H * .5
-            elif 36 < outTemp_F and outTemp_F <= 48:
+            elif 36 < outTemp_F <= 48:
                 chill_time = interval_H
-            elif 48 < outTemp_F and outTemp_F <= 54:
+            elif 48 < outTemp_F <= 54:
                 chill_time = interval_H * .5
-            elif 54 < outTemp_F and outTemp_F <= 60:
+            elif 54 < outTemp_F <= 60:
                 chill_time = 0
-            elif 60 < outTemp_F and outTemp_F <= 65:
+            elif 60 < outTemp_F <= 65:
                 chill_time = interval_H * -.5
             elif 65 < outTemp_F:
                 chill_time = interval_H * -1
-            chill_vt = ValueTuple(chill_time, 'hour', 'group_elapsed')
+            else:
+                chill_time = 0
             log.debug("Found Utah chill hours of %s hours", chill_time)
         elif self.algorithm == 'modified':
             # Use the "Modified" algorithm, counting only chill temps between 32 and 45F
-            if outTemp_F < 45 and 32 < outTemp_F:
+            if 32 < outTemp_F < 45:
                 chill_time = interval_H
             else:
                 chill_time = 0
-            chill_vt = ValueTuple(chill_time, 'hour', 'group_elapsed')
             log.debug("Found Modified chill hours of %s hours", chill_time)
         else:
-            # Don't recognize the exception. Fail hard:
-            raise ValueError(self.algorithm)
+            # Don't recognize the algorithm type. Fail hard:
+            raise ValueError("Unrecognized chill time algorithm '%s'" % self.algorithm)
+
+        # Form a ValueTuple out of our results and return it
+        chill_vt = ValueTuple(chill_time, 'hour', 'group_elapsed')
```

Comment:
Using new group_duration
```suggestion
        chill_vt = ValueTuple(chill_time, 'hour', 'group_duration')
```